tests/codegen-llvm/annotate-moves/call-arg-scope.rs

```diff
@@ -0,0 +1,33 @@
+//@ compile-flags: -Z annotate-moves=8 -Copt-level=0 -g
+
+#![crate_type = "lib"]
+
+#[derive(Clone, Copy)]
+pub struct LargeStruct {
+    pub data: [u64; 20], // 160 bytes
+}
+
+// This test verifies that when passing arguments to functions, the actual CALL instruction
+// does not have the compiler_move debug scope, even though the argument itself might be
+// annotated with compiler_move in MIR.
+//
+// Note: On most ABIs, large structs are passed by pointer even when written as "by value",
+// so there may not be an actual memcpy operation to attach compiler_move to. This test
+// mainly verifies that IF debug info is emitted, the call itself uses the source location.
```

Comment:
It'd be nice with a test that does something like:
```rust
//@ revisions: by_pointer by_value
// Some target with an ABI that passes by pointer
//@ [by_pointer] compile-flags: --target x86_64-unknown-linux-gnu
//@ [by_pointer] needs-llvm-components: x86_64
// Some target with an ABI that passes by value (dunny one on top of my head)
//@ [by_value] compile-flags: --target abc-vendor-os
//@ [by_value] needs-llvm-components: abc
```

That would give me more confidence in that we're actually testing the right thing.